### PR TITLE
Fix Copilot option bug

### PR DESCRIPTION
**Pull request by akhera99**
Fixes bug where default option value was not being respected

### CODE REVIEW

File: src/VisualStudio/CSharp/Impl/Options/CSharpVisualStudioCopilotOptionsService.cs [hunk 1]
```diff
@@ -77,7 +77,7 @@ public async Task<bool> IsCopilotOptionEnabledAsync(string optionName)
         var settingManager = await _settingsManagerTask.ConfigureAwait(false);
         // The bool setting is persisted as 0=None, 1=True, 2=False, so it needs to be retrieved as an int.
         return settingManager.TryGetValue($"{CopilotOptionNamePrefix}.{optionName}", out int isEnabled) == GetValueResult.Success
-            && isEnabled == 1;
+            && isEnabled != 2;
```
Comment (Cosifne) [inline src/VisualStudio/CSharp/Impl/Options/CSharpVisualStudioCopilotOptionsService.cs:80]: Any context?

Comment (Cosifne) [reply src/VisualStudio/CSharp/Impl/Options/CSharpVisualStudioCopilotOptionsService.cs:80]: Oh nevermind I see the comment.

Comment (genlu) [reply src/VisualStudio/CSharp/Impl/Options/CSharpVisualStudioCopilotOptionsService.cs:80]: yea, this is now enabled by default (the source of truth is in copilot repo, where all those options are defined)

**Review by genlu — APPROVED** [review]

File: src/VisualStudio/CSharp/Impl/Options/CSharpVisualStudioCopilotOptionsService.cs (hunk 1 above)
Comment (JoeRobich) [reply src/VisualStudio/CSharp/Impl/Options/CSharpVisualStudioCopilotOptionsService.cs:80]: Copilot needs an option provider service so they can apply the defaults when you request values.

Comment (genlu) [reply src/VisualStudio/CSharp/Impl/Options/CSharpVisualStudioCopilotOptionsService.cs:80]: do you have an example for that?

Comment (JoeRobich) [reply src/VisualStudio/CSharp/Impl/Options/CSharpVisualStudioCopilotOptionsService.cs:80]: No, it was all me wishing something existed. They could write the defaults for their settings as `$"{CopilotOptionNamePrefix}.{option.Name}.Default"` or something similar. Then we could defer to whatever they ship.

Comment (tmat) [reply src/VisualStudio/CSharp/Impl/Options/CSharpVisualStudioCopilotOptionsService.cs:80]: Why are we not using our existing options infra?

**Pull request merged**